components/dashboard/src/service/service.ts

```diff
@@ -49,22 +38,38 @@ function createGitpodService<C extends GitpodClient, S extends GitpodServer>() {
             }
         });
 
-        if (_websocket && "reconnect" in _websocket) {
-            reconnect = () => { (_websocket as any).reconnect() };
-        }
     // }
     const service = new GitpodServiceImpl<C, S>(proxy);
-    (service as any).reconnect = reconnect;
+    (service as any).reconnect = () => {
+        const ws = (webSocket as any);
+        if (typeof ws.reconnect === "function") {
+            ws.reconnect();
+        } else {
+            console.log("WebSocket reconnect not possible.");
+        }
+    };
     return service;
 }
 
 declare global {
-    type GitpodService = ReturnType<typeof createGitpodService> & { reconnect: () => void }
-    interface Window { gitpodService?: GitpodService; }
+    interface Window {
```

Comment:
I'm still not a fan of typing this :-)

Reply:
can you give an example, how to access it then? 

maybe exporting a getter function here?